src/Catalog.jl

```diff
@@ -0,0 +1,137 @@
+module Catalog
+
+import NCDatasets
+
+import Base: get
+import ..Sim: available_vars
+import ..Var: OutputVar
+
+export NCCatalog, add_file!, available_vars
+
+"""
+    Organize external NetCDF files to easily initialize `OutputVar`s.
+"""
+struct NCCatalog{DOI <: Dict, DOV <: Dict}
+    "Dictionary that maps aliases to tuples of file path and variable name in the NetCDF
+    file. The identifier can either be the variable name or an alias for the variable name
+    in the file."
+    identifier_to_path_varname::DOI
+
+    "Dictionary that maps variable names to filepaths. This is a fallback if a variable is
+    requested and cannot be found in `identifier_to_path_varname`."
+    varname_to_filepath::DOV
+end
+
+"""
+    NCCatalog()
+
+Construct an empty `NCCatalog`.
+
+Files can be added with [`ClimaAnalysis.Catalog.add_file!`](@ref).
+"""
+function NCCatalog()
+    identifier_to_path_varname = Dict{String, Tuple{String, String}}()
+    varname_to_filepath = Dict{String, String}()
+    return NCCatalog(identifier_to_path_varname, varname_to_filepath)
+end
+
+"""
+    add_file!(catalog::NCCatalog, filepath, short_names...)
+
+Add the file at `filepath` to `catalog` with `short_names`.
+
+The argument `short_names` can either be a short name (e.g., `"rsdt"`) or a pair of strings
+(e.g., `"solar_mon" => "rsdt"`). If a pair is used, then the first name is the variable name
+in the NetCDF file and the second name is the alias which will be used for `get`.
+
+!!! note "Multiple files with the same short names"
+    If there are multiple files that contain the same short names, then you should pass in
+    the desired file first or pass in a short name or a pair of short name to alias for
+    `short_names` to make the variable in the file retrievable with `get`.
+"""
+function add_file!(catalog::NCCatalog, filepath, short_names...)
+    isfile(filepath) || error("$filepath does not exist")
```

Comment:
Nitpick for a slightly clearer error:
```suggestion
    throw(ArgumentError("NetCDF file not found: $filepath"))
```